### DISCUSSION TITLE
Stop the Tooltip from snapping to `null` value points?

### DISCUSSION THREAD

**Discussion by lunelson** [recharts/recharts#4228] [Q&A]
I have a dataset in which I've inserted some points with null values, so as to not have the `LineChart` connect lines across those spans; however the `Tooltip` still snaps to those points, and this causes a glitchy behaviour in the animation, since it causes the tooltip box to disappear, which re-sets the position of the tooltip box to its default, and then it animates back in from that point when it appears again. 

What I've described can be seen [in the Storybook](https://recharts.org/en-US/storybook) at `Examples > LineChart > Connect Nulls`.

How can I make the `Tooltip` ignore these null points?

**Comment by ckifer** [comment]
@lunelson I don't see a tooltip when hovering over a `null` point either with or without `connectNulls`. Do you mean the vertical line (cursor)?

Could you make a video that explains the issue more clearly

**Reply by lunelson** [reply]
Thanks @ckifer, yeah what I mean is (A) the cursor continues to snap, to the point which is _not really supposed to be there_; but also (B) when you move back to a non-null point, the tooltip can fly in, in a jarring way, because it seems snapping to the null point has zeroed out its `position` value.

What I'd like is for the `Tooltip`—which is to say, both the cursor and the popover elements—to ignore the `null` point completely

https://github.com/recharts/recharts/assets/1242864/f76ad719-ca79-4e18-9694-38443bf676f2

**Reply by ckifer** [reply]
>  is not really supposed to be there

The point is not there and the X Axis value of "Page D" still exists, I think thats why the cursor still has value snap to that point. Could definitely make that configurable though.

>  the tooltip can fly in, in a jarring way

The way that things currently work, this would happen anyways. If I removed the null point snapping in between, you would instead go from the last state (Page E) to Page C in the same animated / jarring manner that you don't like. You can see this if you hover over Page E, bring your mouse down and out of the chart, and bring it back into the chart at Page C.

I understand the issue, but I think the ask would be more to reset the tooltip position back to default if you hover over a null point or exit/re-enter the chart as removing the snapping itself wouldn't help with the problem.

This isn't currently possible though and would need to be a feature request
